hooks/sourcefile/sourcefile.go

```diff
@@ -0,0 +1,38 @@
+package logrus_sourcefile
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"runtime"
+	"strings"
+)
+
+type SourceFileHook struct {
+	LogLevel logrus.Level
+}
+
+func (hook *SourceFileHook) Fire(entry *logrus.Entry) (_ error) {
+	for skip := 4; skip < 9; skip++ {
+		_, file, line, _ := runtime.Caller(skip)
```

Comment:
Yeah, my problem with this type of patch is that this feels very sketchy and error prone for something as critical as a logger.